addons/2d-color-picker/paint-editor.js

```diff
@@ -0,0 +1,210 @@
+    const saColorPicker = document.createElement("div");
+    saColorPicker.className = "sa-2dcolor-picker";
+    saColorPicker.style.background = "#" + convertToGeneralColor(defaultColor || "ff0000");
+
+    const saColorPickerImage = Object.assign(document.createElement("img"), {
+      className: "sa-2dcolor-picker-image",
+      src: addon.self.dir + "/assets/sv-gr.png",
+      draggable: false,
+    });
+    const saColorPickerHandle = Object.assign(document.createElement("div"), {
+      className: "slider_handle_3f0xk",
+    });
+    saColorPickerHandle.style.pointerEvents = "none";
+
+    // create the label
+    const saColorLabel = document.createElement("div");
+    saColorLabel.className = "color-picker_row-header_173LQ sa-2dcolor-label";
+    const saColorLabelName = document.createElement("span");
+    saColorLabelName.className = "color-picker_label-name_17igY sa-2dcolor-label-name";
+    saColorLabelName.innerText = "Shade";
+    const saColorLabelVal = document.createElement("span");
+    saColorLabelVal.className = "color-picker_label-readout_9vjb2 sa-2dcolor-label-val";
+    saColorLabel.appendChild(saColorLabelName);
+    saColorLabel.appendChild(saColorLabelVal);
+
+    function updateHandle(e, keyPressed, originalPos) {
+      let cx = Math.min(Math.max(e.clientX - saColorPicker.getBoundingClientRect().x, 0), 150);
+      let cy = Math.min(Math.max(e.clientY - saColorPicker.getBoundingClientRect().y, 0), 150);
+      if (keyPressed == 16) {
+        if (Math.abs(cx - originalPos.x) > Math.abs(cy - originalPos.y)) cy = originalPos.y;
+        else cx = originalPos.x;
+      }
+      saColorPickerHandle.style.left = cx - 8 + "px";
+      saColorPickerHandle.style.top = cy - 8 + "px";
+      saColorLabelVal.innerText = `${Math.round((cx / 150) * 100)}, ${100 - Math.round((cy / 150) * 100)}`;
+    }
+
+    function updateHandleFinal(s, v) {
+      saColorPickerHandle.style.left = s * 150 - 8 + "px";
+      saColorPickerHandle.style.top = (1 - v) * 150 - 8 + "px";
+      saColorLabelVal.innerText = `${Math.round(s * 100)}, ${Math.round(v * 100)}`;
+    }
+
+    function updateFinal(e, keyPressed, originalPos) {
+      rateLimiter.limit(() => {
+        let ox = Math.min(Math.max(e.clientX - saColorPicker.getBoundingClientRect().x, 0), 150);
+        let oy = Math.min(Math.max(e.clientY - saColorPicker.getBoundingClientRect().y, 0), 150);
+        if (keyPressed == 16) {
+          if (Math.abs(ox - originalPos.x) > Math.abs(oy - originalPos.y)) oy = originalPos.y;
+          else ox = originalPos.x;
+        }
+
+        let color = tinycolor(getColor(element)).toHsv();
+        let s = ox / 150;
+        let v = 1 - oy / 150;
+        let newColor = tinycolor({ h: color.h, s: s, v: v }).toHex();
+        setColor(newColor, element);
+        updateHandleFinal(s, v);
+      });
+
+      window.onpointermove = null;
+      window.onpointerup = null;
+      window.onkeydown = null;
+      window.onkeyup = null;
```

Comment:
Is there a reason for this?

Reply:
reset the events, just to be safe!

Reply:
Safe from what? Does it not work if you don't do this?